src/netbase.cpp

```diff
@@ -958,3 +1075,131 @@ void InterruptSocks5(bool interrupt)
 {
     interruptSocks5Recv = interrupt;
 }
+
+
+static inline bool IOErrorIsPermanent(int err)
+{
+    return err != WSAEAGAIN && err != WSAEWOULDBLOCK && err != WSAEINTR && err != WSAEINPROGRESS;
```

Comment:
Build warning here, logical AND of equal expressions [-Wlogical-op]

Reply:
Is this on Windows? Which compiler version? Is it upset that `WSAEAGAIN` equals to `WSAEWOULDBLOCK` and so we end up doing something like `err != 5 && err != 5`?

I checked that gcc 10.2.1, 9.3.0 and 8.4.0 don't emit the warning if we do `err != 5 && err != 6 && err != 5`, so I just reordered the expressions as a simple fixup ro this. Will include in the next push.

Reply:
gcc Debian 10.2.1

Reply:
Fixed in f0577c4?

Reply:
> Fixed in [f0577c4](https://github.com/bitcoin/bitcoin/commit/f0577c4d722c580c48ba1734c4b996763cc44d12)?

Yes

Reply:
Yes, thank you -- tested that it is fixed in 188ba34 and in 2ae504c